xdsl/ir/core.py

```diff
@@ -387,6 +388,11 @@ def __eq__(self, other: object) -> bool:
         return self is other
 
 
+Operand: TypeAlias = SSAValue
+VarOperand: TypeAlias = list[Operand]
```

Comment:
It feels weird to me that this is now moved to `core.py`, where these are `irdl` concepts?
Did you have to do this because of cycle of reference? If that's the case, then that's fine.

Reply:
It's more sneaky than this;

Type hints are for typecheckers only, right? So, as a Python object, Operand actually *is* SSAValue.
Which means `Operand.__module__` actually *is* `SSAValue.__module__` AKA `xdsl.ir.core`.

So, when generating their import statements in stub, it would need special-casing to do it right, or just resort back to type them `SSAValue` on `IRDLOperation`s, which sounds too bad :thinking: 

Seeing it from that point of view, having `Operand` defined in `xdsl.irdl.core` to end up having `__module__` as `xdsl.ir.core` is what felt weird to me.

Now that I mention it though, it solves half the problem, 'cause I'll have to special case `VarOperand` anyway, which as `list` would have `builtins` as `__module__` :smiling_face_with_tear: So, I'll think about it and maybe just make it less fancy on the type stub side..

Reply:
Reverted the moving around, and solved inside dialect_stub.py